openfl-tutorials/interactive_api/PyTorch_DogsCats_ViT/envoy/dogs_cats_shard_descriptor.py

```diff
@@ -0,0 +1,188 @@
+            print('Your dataset is absent or damaged. Downloading ... ')
+            api = KaggleApi()
+            api.authenticate()
+
+            if os.path.exists('data/train'):
+                shutil.rmtree('data/train')
+            if os.path.exists('data/test'):
+                shutil.rmtree('data/test')
+
+            api.competition_download_file(
+                'dogs-vs-cats-redux-kernels-edition',
+                'train.zip', path=self.data_folder
+            )
+            api.competition_download_file(
+                'dogs-vs-cats-redux-kernels-edition',
+                'test.zip', path=self.data_folder
+            )
+            with ZipFile(self.data_folder / 'train.zip', 'r') as zipobj:
+                zipobj.extractall(self.data_folder)
+
+            os.remove(self.data_folder / 'train.zip')
+
+            with ZipFile(self.data_folder / 'test.zip', 'r') as zipobj:
+                zipobj.extractall(self.data_folder)
+
+            os.remove(self.data_folder / 'test.zip')
+
+            DogsCatsShardDescriptor.save_all_md5(self.data_folder)
+
+        # Settings for resizing data
+        self.enforce_image_hw = None
+        if enforce_image_hw is not None:
+            self.enforce_image_hw = tuple(map(int, enforce_image_hw.split(',')))
+
+        # Calculating data and target shapes
+        ds = self.get_dataset()
+        sample, target = ds[0]
+        self._sample_shape = [str(dim) for dim in sample.shape]
+        self._target_shape = [str(*target.shape)]
+
+        assert self._target_shape[0] == '1', 'Target shape Error'
+
+    def get_dataset(self, dataset_type='train'):
+        """Return a shard dataset by type."""
+        return DogsCatsShardDataset(
+            data_type=dataset_type,
+            dataset_dir=self.data_folder,
+            rank=self.rank,
+            worldsize=self.worldsize,
+            enforce_image_hw=self.enforce_image_hw
+        )
+
+    @staticmethod
+    def calc_all_md5(data_folder):
+        """Calculate hash of all dataset."""
+        md5_dict = {}
+        for root, _, files in os.walk(data_folder):
+            for file in files:
+                if file == 'dataset.md5':
+                    continue
+                md5_calc = md5()
+                rel_dir = os.path.relpath(root, data_folder)
+                rel_file = os.path.join(rel_dir, file)
+
+                with open(data_folder / rel_file, 'rb') as f:
+                    for chunk in iter(lambda: f.read(4096), b''):
+                        md5_calc.update(chunk)
+                    md5_dict[rel_file] = md5_calc.hexdigest()
+        return md5_dict
+
+    @staticmethod
+    def save_all_md5(data_folder):
+        """Save dataset hash."""
+        all_md5 = DogsCatsShardDescriptor.calc_all_md5(data_folder)
+        with open(os.path.join(data_folder, 'dataset.md5'), 'w') as f:
+            json.dump(all_md5, f)
+
+    @staticmethod
+    def check_dataset(data_folder):
```

Comment:
Maybe better rename to something starts with `is_`. `is_dataset_complete` for example.